src/utils/fluent.js

```diff
@@ -113,11 +114,17 @@ function getMessageWithLocale (id, localePreferences, args) {
 
   if (!bundle.hasMessage(id)) bundle = fluentBundles.en
 
-  if (bundle.hasMessage(id)) return bundle.formatPattern(bundle.getMessage(id).value, args)
+  if (bundle.hasMessage(id)) {
+    // @ts-ignore We know `.value` exists thanks to `.hasMessage()`
```

Comment:
Unfortunately no; you can tell TS to narrow the type of a particular variable so that it can no longer be `undefined`, but not for a particular function invocation with a specific parameter. (That would quickly make type checking very expensive, I think.) In regular TypeScript I'd use a non-null assertion here, but that doesn't seem possible in JSDoc mode.

I'm not a particularly big fan of adding a `@ts-ignore` here (mostly because I don't want it to become one of the first things people reach for), but in this case, it doesn't introduce unsafety, so I think it's OK.

However, your comment did remind me that `@ts-ignore` should really be considered an escape hatch, and thus always accompanied by a comment explaining why it's needed. So I added [the ESLint TS plugin](https://typescript-eslint.io/) and turned on [the rule that checks for that](https://typescript-eslint.io/rules/ban-ts-comment/): 118ab77f9

Reply:
Ah, the TS plugin does find a whole bunch of errors in our deprecated code. I updated our ESLint command to just no longer lint that, since we're going to remove it anyway: 85edfe12b